lms/resources/lti_launch.py

```diff
@@ -29,11 +29,15 @@ def __init__(self, request):
     @cached_property
     def course(self):
         """Get the course this LTI launch based on the request's params."""
+        if existing_course := self._request.find_service(
+            name="course"
+        ).get_by_context_id(self._request.parsed_params["context_id"]):
+            return existing_course
 
         return self._request.find_service(name="course").upsert_course(
```

Comment:
If that fails, create it with the initial extra.

This call here could be to an hypothetical `create` but `upsert` is the method we have.